### PR TITLE
[BACKLOG-41229] - Decode PVFS File Paths: PDI - Browse - Select a file from VFS

### PR DESCRIPTION
**Pull request by nawaz34-hitachivantara**
Jira bug - https://hv-eng.atlassian.net/browse/BACKLOG-41229.

Attachment containing changes after code fix : 

image

### CODE REVIEW

**Comment by buildguy** [comment]
<div align='center'>

[👍 Frogbot scanned this pull request and found that it did not add vulnerable dependencies.](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

<details>
<summary> <b>Note:</b> </summary>


---
<div align='center'>

**Frogbot** also supports **Contextual Analysis, Secret Detection, IaC and SAST Vulnerabilities Scanning**. This features are included as part of the [JFrog Advanced Security](https://jfrog.com/advanced-security) package, which isn't enabled on your system.

</div>


</details>


---
<div align='center'>

[🐸 JFrog Frogbot](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

File: plugins/file-open-save-new/api/src/main/java/org/pentaho/di/plugins/fileopensave/api/providers/Entity.java [hunk 1]
```diff
@@ -37,4 +39,14 @@ public interface Entity {
...
   boolean isCanDelete();
   EntityType getEntityType();
+
+  /**
+   * Gets the decoded version of name of an element.
+   * Defaults to getName() {@code String}.
+   *
+   * @return A non-empty name of an element.
+   */
+  default String getNameDecoded() throws FileSystemException {
```
Comment (dcleao) [inline plugins/file-open-save-new/api/src/main/java/org/pentaho/di/plugins/fileopensave/api/providers/Entity.java:49]: We should try to keep Apache VFS types as an implementation detail.
You can use the existing exception, `org.pentaho.di.plugins.fileopensave.api.providers.exception.FileException` instead.

File: plugins/file-open-save-new/api/src/main/java/org/pentaho/di/plugins/fileopensave/api/providers/Entity.java [hunk 2]
```diff
@@ -37,4 +39,14 @@ public interface Entity {
   boolean isCanEdit();
   boolean isCanDelete();
   EntityType getEntityType();
+
+  /**
+   * Gets the decoded version of name of an element.
+   * Defaults to getName() {@code String}.
```
Comment (dcleao) [inline plugins/file-open-save-new/api/src/main/java/org/pentaho/di/plugins/fileopensave/api/providers/Entity.java:45]: Should be instead `Defaults to {@link #getName()}.`

File: plugins/file-open-save-new/api/src/main/java/org/pentaho/di/plugins/fileopensave/api/providers/Tree.java [hunk 3]
```diff
@@ -46,4 +48,12 @@ default boolean isCanDelete() {
...
     return EntityType.TREE;
   }
+
+  /**
+   * Gets the decoded version of name of an element.
+   * Defaults to getName() {@code String}.
+   *
+   * @return A non-empty name of an element.
+   */
+  default String getNameDecoded() throws FileSystemException { return getName(); }
```
Comment (dcleao) [inline plugins/file-open-save-new/api/src/main/java/org/pentaho/di/plugins/fileopensave/api/providers/Tree.java:58]: Same remarks as for `Entity`.

File: plugins/file-open-save-new/core/src/main/java/org/pentaho/di/plugins/fileopensave/providers/vfs/model/VFSFile.java [hunk 4]
```diff
@@ -186,4 +188,12 @@ public int hashCode() {
   public EntityType getEntityType(){
     return EntityType.VFS_FILE;
   }
+
+  @Override
+  public String getNameDecoded() throws FileSystemException {
+    if ( StringUtils.isNotEmpty( getName() ) ) {
+      return UriParser.decode( getName() );
+    }
+    return getName();
```
Comment (dcleao) [inline plugins/file-open-save-new/core/src/main/java/org/pentaho/di/plugins/fileopensave/providers/vfs/model/VFSFile.java:197]: Catch ApacheVFS `FileSystemException` and wrap it in a `org.pentaho.di.plugins.fileopensave.api.providers.exception.FileException`.

**Review by dcleao — CHANGES_REQUESTED** [review]

File: plugins/file-open-save-new/api/src/main/java/org/pentaho/di/plugins/fileopensave/api/providers/Entity.java (hunk 1 above)
Comment (nawaz34-hitachivantara) [reply plugins/file-open-save-new/api/src/main/java/org/pentaho/di/plugins/fileopensave/api/providers/Entity.java]: Addressed

File: plugins/file-open-save-new/api/src/main/java/org/pentaho/di/plugins/fileopensave/api/providers/Entity.java (hunk 2 above)
Comment (nawaz34-hitachivantara) [reply plugins/file-open-save-new/api/src/main/java/org/pentaho/di/plugins/fileopensave/api/providers/Entity.java]: Addressed.

File: plugins/file-open-save-new/api/src/main/java/org/pentaho/di/plugins/fileopensave/api/providers/Tree.java (hunk 3 above)
Comment (nawaz34-hitachivantara) [reply plugins/file-open-save-new/api/src/main/java/org/pentaho/di/plugins/fileopensave/api/providers/Tree.java]: Addressed.

**Comment by buildguy** [comment]
<div align='center'>

[👍 Frogbot scanned this pull request and found that it did not add vulnerable dependencies.](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

<details>
<summary> <b>Note:</b> </summary>


---
<div align='center'>

**Frogbot** also supports **Contextual Analysis, Secret Detection, IaC and SAST Vulnerabilities Scanning**. This features are included as part of the [JFrog Advanced Security](https://jfrog.com/advanced-security) package, which isn't enabled on your system.

</div>


</details>


---
<div align='center'>

[🐸 JFrog Frogbot](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 7m 19s
Build command:
```console
mvn clean verify -B -e -Daudit -Djs.no.sandbox -pl \
plugins/file-open-save-new/api,plugins/file-open-save-new/core
```
---
:ok_hand: All tests passed!
 
Tests run: **66**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/pdi/job/PR-9450/2/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
<div align='center'>

[👍 Frogbot scanned this pull request and found that it did not add vulnerable dependencies.](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

<details>
<summary> <b>Note:</b> </summary>


---
<div align='center'>

**Frogbot** also supports **Contextual Analysis, Secret Detection, IaC and SAST Vulnerabilities Scanning**. This features are included as part of the [JFrog Advanced Security](https://jfrog.com/advanced-security) package, which isn't enabled on your system.

</div>


</details>


---
<div align='center'>

[🐸 JFrog Frogbot](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 7m 21s
Build command:
```console
mvn clean verify -B -e -Daudit -Djs.no.sandbox -pl \
plugins/file-open-save-new/api,plugins/file-open-save-new/core
```
---
:ok_hand: All tests passed!
 
Tests run: **66**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/pdi/job/PR-9450/3/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Review by dcleao — APPROVED** [review]

**Review by dcleao — APPROVED** [review]

**Comment by buildguy** [comment]
<div align='center'>

[👍 Frogbot scanned this pull request and found that it did not add vulnerable dependencies.](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

<details>
<summary> <b>Note:</b> </summary>


---
<div align='center'>

**Frogbot** also supports **Contextual Analysis, Secret Detection, IaC and SAST Vulnerabilities Scanning**. This features are included as part of the [JFrog Advanced Security](https://jfrog.com/advanced-security) package, which isn't enabled on your system.

</div>


</details>


---
<div align='center'>

[🐸 JFrog Frogbot](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 23m 23s
Build command:
```console
mvn clean verify -B -e -Daudit -Djs.no.sandbox -pl \
plugins/file-open-save-new/api,plugins/file-open-save-new/core
```
---
:ok_hand: All tests passed!
 
Tests run: **66**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/pdi/job/PR-9450/4/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
<div align='center'>

[👍 Frogbot scanned this pull request and found that it did not add vulnerable dependencies.](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

<details>
<summary> <b>Note:</b> </summary>


---
<div align='center'>

**Frogbot** also supports **Contextual Analysis, Secret Detection, IaC and SAST Vulnerabilities Scanning**. This features are included as part of the [JFrog Advanced Security](https://jfrog.com/advanced-security) package, which isn't enabled on your system.

</div>


</details>


---
<div align='center'>

[🐸 JFrog Frogbot](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 7m 34s
Build command:
```console
mvn clean verify -B -e -Daudit -Djs.no.sandbox -pl \
plugins/file-open-save-new/api,plugins/file-open-save-new/core
```
---
:ok_hand: All tests passed!
 
Tests run: **66**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/pdi/job/PR-9450/5/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 29m 39s
Build command:
```console
mvn clean verify -B -e -Daudit -Djs.no.sandbox -pl \
core,plugins/file-open-save-new/api,plugins/file-open-save-new/core
```
---
:ok_hand: All tests passed!
 
Tests run: **1385**, Failures: **0**, Skipped: **1** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/pdi/job/PR-9450/6/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 7m 36s
Build command:
```console
mvn clean verify -B -e -Daudit -Djs.no.sandbox -pl \
plugins/file-open-save-new/api,plugins/file-open-save-new/core
```
---
:ok_hand: All tests passed!
 
Tests run: **66**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/pdi/job/PR-9450/7/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
<div align='center'>

[👍 Frogbot scanned this pull request and found that it did not add vulnerable dependencies.](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

<details>
<summary> <b>Note:</b> </summary>


---
<div align='center'>

**Frogbot** also supports **Contextual Analysis, Secret Detection, IaC and SAST Vulnerabilities Scanning**. This features are included as part of the [JFrog Advanced Security](https://jfrog.com/advanced-security) package, which isn't enabled on your system.

</div>


</details>


---
<div align='center'>

[🐸 JFrog Frogbot](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 24m 46s
Build command:
```console
mvn clean verify -B -e -Daudit -Djs.no.sandbox -pl \
plugins/file-open-save-new/api,plugins/file-open-save-new/core
```
---
:ok_hand: All tests passed!
 
Tests run: **66**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/pdi/job/PR-9450/8/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Review by rmansoor — APPROVED** [review]

**Pull request merged**